packages/snap/src/index.ts

```diff
@@ -143,29 +141,62 @@ async function sendTransaction(tx: CeloTransactionRequest, wallet: CeloWallet) {
   return txResponse.wait()
 }
 
-async function getBIP44Node(index: number = 0): Promise<BIP44Node> {
-  // Get the bip44 node corresponding to the path m/44'/52752'/0'/0
-  const bip44Node = await snap.request({
+async function getKeyPair(snap: SnapsGlobalObject, address?: string, addressIndex: number = 0): Promise<KeyPair> {
+  const derivationPath = "m/44'/60'/0'/0"; //Todo - read from state config
+  let extendedPrivateKey;
+  let _addressIndex = 0;
+  const MAX_SEARCH_DEPTH = 50;
+
+  const [, , coinType, account, change] = derivationPath.split('/');
+  const bip44Code = coinType.replace("'", '');
+
+  const bip44Node = (await snap.request({
     method: 'snap_getBip44Entropy',
     params: {
-      coinType: 52752, // https://github.com/satoshilabs/slips/blob/master/slip-0044.md
+      coinType: Number(bip44Code),
     },
-  })
-  const deriveAddress = await getBIP44AddressKeyDeriver(bip44Node)
-  // Derive account w index 
-  return deriveAddress(index)
-}
+  })) as JsonBIP44CoinTypeNode;
 
-async function getPrivateKey(bip44Node?: BIP44Node, index: number = 0): Promise<string> {
-  if (!bip44Node) {
-    bip44Node = await getBIP44Node(index)
-  }
-  if (!bip44Node.privateKey) {
-    throw new Error('Private key is undefined. BIP-44 node is public.')
+  const addressKeyDeriver = await getBIP44AddressKeyDeriver(bip44Node, {
+    account: parseInt(account),
+    change: parseInt(change),
+  }); 
+
+  if (address) {
+    let search;
+    do {
```

Comment:
is the `do` `while` loop aync aware?  some js loops work right with await others do not. afaik

Reply:
I think the problem only arises with functional "loops"/iterators like `.forEach()`. Native language loops in an `async` context should properly `await`.

Reply:
Yes works well. thanks both!